src/wget.c

```diff
@@ -340,6 +340,14 @@ const char * G_GNUC_WGET_NONNULL_ALL get_local_filename(wget_iri_t *iri)
 	} else {
 		fname = wget_iri_get_filename(iri, &buf, config.local_encoding);
 	}
+	
+	if(config.cut_file_get_vars) {
```

Comment:
If config.cut_file_get_vars is set, you can skip the call wget_iri_get_query_as_filename() a few lines above (fname = buf.data instead).
Also you could call wget_iri_get_path() instead of calling wget_iri_get_filename().
That just avoids appending the query to the filename.